_drafts/autolab-and-docker.markdown

```diff
@@ -0,0 +1,56 @@
+---
+layout: post
+title: "Autolab and Docker"
+author: Mihir Pandya
+
+tags: [autograding, docker, tango]
+---
+
+![]({{site.baseurl}}/assets/docker.svg)
+
+As highlighted in our ['Hello World' post][hello-world], we at Autolab want to make Autograding for All a reality. In pursuit of this vision, we started by making Autolab robust enough for a variety of classes. Following that, we [open-sourced][gh-autolab] this project after it had been tried and tested at Carnegie Mellon University. 
+
+While these events have been instrumental in making Autolab usable, the time and effort that needs to be spent in setting up and configuring Tango, the autograding backend, is non-trivial. Furthermore, Tango needs a platform on which it can run jobs. At present, it supports a distributed CMU-specific platform (called Tashi). It also allows jobs to run locally on a single server or laptop, or even a cluster of EC2 machines. However, moving forward we wanted to provide more options that simplified the process of developing, testing, and deploying Tango. Hence, as part of the latest efforts in making Autograding for All a reality, we are integrating Docker with Tango in the [v1.1.0 release][gh-docker] in order to significantly reduce the overhead in getting started and making autograding platform-agnostic.
+
+This post starts by shedding some light on the concepts Tango uses for autograding. Building on those concepts, this post will then outline how Docker is integrated with Tango.
+
+# What it takes to Autograde
+
+A guarantee that Tango must provide is that the environment in which a job runs remains homogeneous between jobs. Hence, Tango executes these jobs on virtual machines that are booted with user-specified images. In order to run jobs on a particular platform such as Amazon EC2, OpenStack or even RackSpace, the Virtual Machine Management (VMM) interface must be implemented for that platform. The [VMM interface][gh-sample-code] is as follows:
```

Comment:
1) Are the necessarily "virtual machines" ? Dockers, what the post is about, are not VMs.
2) "user-specified images" or user-specified environments? Image sounds like very specific term here.


But what does "Virtual Machine Management (VMM) interface", maybe there should be a sentence-long description of what these functions achieve in general.